masonry/src/event_loop_runner.rs

```diff
@@ -430,13 +428,13 @@ impl MainState<'_> {
                     // TODO
                 }
                 render_root::RenderRootSignal::TextFieldFocused => {
-                    // TODO
+                    window.set_ime_allowed(true);
                 }
                 render_root::RenderRootSignal::ImeStarted => {
-                    // TODO
+                    window.set_ime_allowed(true);
                 }
-                render_root::RenderRootSignal::ImeMoved => {
-                    // TODO
+                render_root::RenderRootSignal::ImeMoved(position, size) => {
+                    window.set_ime_cursor_area(position, size);
                 }
```

Comment:
We may want to remove some of these events. They're placeholders, it's not like I had a plan in mind for what IME would look like when I created them.

Reply:
I tried not to mark any of this as YAGNI because I know you have thought a lot about _other_ things in here. It definitely didn't seem like all of these should exist. I'll cut some of them out.

Reply:
I've gone ahead and cleaned these up a bit. Feel free to bikeshed the names a bit.